File: server/models/caso.py
"""
Maia Platform — Case (Processo) Models
"""
from pydantic import BaseModel, Field
from typing import Optional, Any
from enum import Enum


class CasoStatus(str, Enum):
    ATIVO = "ativo"
    EM_ANDAMENTO = "em_andamento"
    SUSPENSO = "suspenso"
    ARQUIVADO = "arquivado"
    ENCERRADO = "encerrado"


class CasoTipo(str, Enum):
    CIVEL = "civel"
    TRABALHISTA = "trabalhista"
    CRIMINAL = "criminal"
    TRIBUTARIO = "tributario"
    ADMINISTRATIVO = "administrativo"
    PREVIDENCIARIO = "previdenciario"
    OUTRO = "outro"


class CasoCreateRequest(BaseModel):
    numero: str = Field(..., min_length=1, max_length=30)
    titulo: str = Field(..., min_length=2, max_length=200)
    tipo: CasoTipo = CasoTipo.CIVEL
    status: CasoStatus = CasoStatus.EM_ANDAMENTO
    cliente_id: Optional[str] = None
    responsavel_id: Optional[str] = None
    tribunal: Optional[str] = None
    vara: Optional[str] = None
    descricao: Optional[str] = None


class CasoUpdateRequest(BaseModel):
    titulo: Optional[str] = None
    tipo: Optional[CasoTipo] = None
    status: Optional[CasoStatus] = None
    cliente_id: Optional[str] = None
    responsavel_id: Optional[str] = None
    tribunal: Optional[str] = None
    vara: Optional[str] = None
    descricao: Optional[str] = None


class CasoResponse(BaseModel):
    id: str
    numero: str
    titulo: str
    tipo: str
    status: str
    cliente_id: Optional[str] = None
    cliente_nome: Optional[str] = None
    responsavel_id: Optional[str] = None
    tribunal: Optional[str] = None
    vara: Optional[str] = None
    descricao: Optional[str] = None
    # Datajud enrichment fields
    classe_processual: Optional[str] = None
    orgao_julgador: Optional[str] = None
    data_ajuizamento: Optional[str] = None
    assuntos: Optional[list[str]] = None
    dados_tribunal: Optional[dict[str, Any]] = None
    predictive_analytics: Optional[dict[str, Any]] = None
    workspace_id: str
    created_at: str
    updated_at: str

File: client/src/types/caso.ts
export type CasoStatus = 'ativo' | 'em_andamento' | 'suspenso' | 'arquivado' | 'encerrado';
export type CasoTipo = 'civel' | 'trabalhista' | 'criminal' | 'tributario' | 'administrativo' | 'previdenciario' | 'outro';

export interface Caso {
    id: string;
    numero: string;
    titulo: string;
    tipo: CasoTipo;
    status: CasoStatus;
    cliente_id?: string;
    cliente_nome?: string;
    responsavel_id?: string;
    tribunal?: string;
    vara?: string;
    descricao?: string;
    workspace_id: string;
    created_at: string;
    updated_at: string;
}

export interface CasoCreateRequest {
    numero: string;
    titulo: string;
    tipo: CasoTipo;
    status: CasoStatus;
    cliente_id?: string;
    tribunal?: string;
    vara?: string;
    descricao?: string;
}

export const STATUS_LABELS: Record<CasoStatus, string> = {
    ativo: 'Ativo',
    em_andamento: 'Em Andamento',
    suspenso: 'Suspenso',
    arquivado: 'Arquivado',
    encerrado: 'Encerrado',
};

export const TIPO_LABELS: Record<CasoTipo, string> = {
    civel: 'Cível',
    trabalhista: 'Trabalhista',
    criminal: 'Criminal',
    tributario: 'Tributário',
    administrativo: 'Administrativo',
    previdenciario: 'Previdenciário',
    outro: 'Outro',
};

export const STATUS_COLORS: Record<CasoStatus, string> = {
    ativo: 'bg-green-500/10 text-green-400 border-green-500/20',
    em_andamento: 'bg-blue-500/10 text-blue-400 border-blue-500/20',
    suspenso: 'bg-yellow-500/10 text-yellow-400 border-yellow-500/20',
    arquivado: 'bg-zinc-500/10 text-zinc-400 border-zinc-500/20',
    encerrado: 'bg-red-500/10 text-red-400 border-red-500/20',
};
